Source/FTPDFAssetRenderer.m

```diff
@@ -1,145 +1,220 @@
 #import "FTPDFAssetRenderer.h"
 
+// From: https://gist.github.com/1209911
+#import <CommonCrypto/CommonDigest.h>
+static NSString * FTPDFMD5String(NSData *input) {
+    unsigned char result[16];
+    CC_MD5(input.bytes, (CC_LONG)input.length, result);
+    return [NSString stringWithFormat:@"%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x",
+            result[0],  result[1],  result[2],  result[3],
+            result[4],  result[5],  result[6],  result[7],
+            result[8],  result[9],  result[10], result[11],
+            result[12], result[13], result[14], result[15]];
+}
+
+
 @implementation FTAssetRenderer (FTPDFAssetRenderer)
 
-+ (FTPDFAssetRenderer *)rendererForPDFNamed:(NSString *)pdfName;
++ (FTPDFAssetRenderer *)rendererForPDFNamed:(NSString *)pdfName
 {
-  NSURL *URL = [[NSBundle mainBundle] URLForResource:pdfName withExtension:@"pdf"];
-  if (URL) {
-    FTPDFAssetRenderer *renderer = [[FTPDFAssetRenderer alloc] initWithURL:URL];
-    return renderer;
-  }
-  return nil;
+    NSURL *URL = [[NSBundle mainBundle] URLForResource:pdfName withExtension:@"pdf"];
+    if (URL != nil) {
+        FTPDFAssetRenderer *renderer = [[FTPDFAssetRenderer alloc] initWithURL:URL];
+        return renderer;
+    }
+
+    return nil;
 }
 
 @end
 
 
 @interface FTPDFAssetRenderer () {
-  CGPDFDocumentRef _document;
+    CGPDFDocumentRef _document;
+    NSString *_cachePathIdentifier;
```

Comment:
These should really just become properties on a private class extension nowadays imo.